src/java.base/share/classes/java/math/BigDecimal.java

```diff
@@ -3141,7 +3141,17 @@ else if (ys != INFLATED)
      * compareTo}, this method considers two {@code BigDecimal}
      * objects equal only if they are equal in value and
      * scale. Therefore 2.0 is not equal to 2.00 when compared by this
-     * method.
+     * method since the former has [{@code BigInteger},
+     * {@code scale}] components equal to [20, 1] while the latter has
+     * components equals to [200, 2].
+     *
+     * <p>One example that shows how 2.0 and 2.00 are <em>not</em>
+     * substitutable for each other under some arithmetic operations
+     * either are the two expressions:<br>
+     * {@code new BigDecimal("2.0" ).divide(BigDecimal.valueOf(3),
+     * HALF_UP)} which evaluates to 0.7 and <br>
+     * {@code new BigDecimal("2.00").divide(BigDecimal.valueOf(3),
+     * HALF_UP)} which evaluates to 0.<b>6</b>7.
      *
```

Comment:
Should this be in an `@apiNote`?

I'm not sure adding the boldface 0.**6**7 is helpful; 0.7 is self-evidently unequal to 0.67.

Reply:
Changed as suggested in final version.